.github/workflows/test-ci-push.yml

```diff
@@ -0,0 +1,36 @@
+name: test-CI-on-push
+
+concurrency:
+  group: ci-${{ github.run_id }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - main
+      - v[0-9]+
+      - v[0-9]+.[0-9]+
+      - cryostat-v[0-9]+.[0-9]+
+
+jobs:
+  get-test-image-tag:
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.compute-tag.outputs.tag }}
+    steps:
+    - name: Compute test image tag
+      id: compute-tag
+      run: |
+        prefix="ci"
+        if [ -n "${{ github.event.number }}" ]; then
+          prefix="pr-${{ github.event.number }}"
+        fi
+        echo "tag=${prefix}-$GITHUB_SHA" >> $GITHUB_OUTPUT
+  
+  run-test-jobs:
+    uses: ./.github/workflows/test-ci-reusable.yml
+    needs: [get-test-image-tag]
+    with:
+      repository: ${{ github.event.pull_request.head.repo.full_name }}
+      ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
I think these `github.event.pull_request` properties are undefined on `push` event, in which case we're getting the Checkout action to use its defaults. These defaults should be the originating repository and ref, which for push is what we want anyways. Maybe for clarity we should make these parameters optional and omit them for the push workflow?